lib/uplink/uplink.go

```diff
@@ -91,6 +94,15 @@ type Uplink struct {
 func NewUplink(ctx context.Context, cfg *Config) (_ *Uplink, err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	return NewUplinkWithLogger(ctx, zap.L(), cfg)
+}
+
+// NewUplinkWithLogger creates a new Uplink with a zap.Logger.
+// This is the first step to create an uplink session with a
+// user specified config or with default config, if nil config
+func NewUplinkWithLogger(ctx context.Context, log *zap.Logger, cfg *Config) (_ *Uplink, err error) {
```

Comment:
can we add a logger option to the config struct instead? that way we wouldn't need a new constructor

Reply:
That simplified things quite a lot. Done :)